tap_zendesk/discover.py

```diff
@@ -20,12 +24,40 @@ def load_shared_schema_refs():
 
     return shared_schema_refs
 
-def discover_streams(client):
+def discover_streams(client, config):
     streams = []
+    error_list = []
     refs = load_shared_schema_refs()
 
+
     for s in STREAMS.values():
-        s = s(client)
+        s = s(client, config)
         schema = singer.resolve_schema_references(s.load_schema(), refs)
+        try:
+            # Here it call the check_access method to check whether stream have read permission or not.
+            # If stream does not have read permission then append that stream name to list and at the end of all streams
+            # raise forbidden error with proper message containinn stream names.
+            s.check_access()
+        except ZendeskForbiddenError as e:
+            error_list.append(s.name) # Append stream name to the
+        except zenpy.lib.exception.APIException as e:
+            err = json.loads(e.args[0]).get('error')
```

Comment:
@prijendev Since `e.args[0]` is being used multiple times. Please make it a variable and get it converted in the `json.loads`. and re-use it further everywhere

Reply:
Updated